### ISSUE TITLE
Обоснование Platformeco

### ISSUE THREAD

**Issue by 61803** [adeo/lmru--tech#85]
1. Нет обоснования использования Platformeco.
2. Нет сравнения с другими вариантами. Например, с Kong.
3. Есть множество нареканий и проблем, связанных с использованием Platformeco.
4. Есть требования, в том числе, по стабильности и производительности, которым Platformeco не удовлетворяет.

**Comment by mahdimahavalm** [comment]
1) Looks like Author has a limited knowledge about LMRU standards, documents, research & comparisons. Recommendation from integration platform to update the knowledge with Enterprise architect of corresponding domain.
2) Looks like author has a lack of expertise in the subject as try to compare this products. . Recommendation from integration platform to request for trainings on the subject areas from integration & application platform (Aleksandr Stavitsky & Aleksandr Kamchatnov - as a contacts for reference)
3) Please provide details and be precise 
4) Same as for points 3. Unless please do not publish unconfirmed informtation.

Regards, 
 LMRU Exchange Platform

**Comment by alem355** [comment]
@61803 Мало данных, дай больше конкретики. по п.3-4 без деталей сложно понять о чем ты конкретно говоришь.
п.1-2 Есть корпоративные стандарты утвержденные в том числе LEA. Можно обратиться к профильной команде - платформе интеграций они расскажут детали.

**Comment by 61803** [comment]
@alem355 Значит мы как инженеры не имеем возможности выбирать удобные и подходящие нам инструменты? Это большой минус для компании, как HR бренда.

Конкретика:

1. Мне как разработчику удобно использовать исходный код и среду разработки исходного кода.
2. Мне хочется иметь возможность делать ревью кода.
3. Мне хочется более консистентный стек языков и технологий в продукте. Сейчас platformeco предлагает только язык Java Script.
4. Мне хочется использовать актуальные для Рынка инструменты.
5. Мне хочется единого подхода (пайплайна) к CI/CD во всех приложениях.
6. Мне хочется иметь правильную с технической точки зрения архитектуру. Сейчас в качестве BFF platformeco в части функционала дублирует Kong.

**Comment by 61803** [comment]
@mahdimahavalm не совсем понял, зачем переходить на английский. Этот проект внутренний для России.

**Comment by mahdimahavalm** [comment]
@61803 
 According to company rule formal communications and standards should be in english, there is no changes so far.
1) it's possible
2) It's completely supported
3) What does it mean consistent? Please be precise and provide business justification
4) Its more than actual for the market
5) It's exactly the case - CI/CD implementation in product s absolutely standart for the company
6) You understanding is not correct. Looks like that you have lack of experience on the topic so please participate in nearest Exchange Platform training


Best regards, 
 Exchanage Platform Team

**Comment by panovst** [comment]
> 5. It's exactly the case - CI/CD implementation in product s absolutely standart for the company

Hello!
as far as I understand, the following is meant here: be able to use the same flow for working with git, however, the Platformeco limits the use of a specific - git flow.
 For example, in our project, the main flow is [OneFlow](https://app.gitbook.com/o/0gRlAkSocTvnN36NTnDZ/s/CbpNZaHwSCmHhyCzbT5Q/studies/ci-cd/git-flows/oneflow)

**Comment by 61803** [comment]
@mahdimahavalm
Прошу указать на документ, регламентирующий общение между коллегами в LMRU на английском.

- Итак. Насчет ревью. Технически это возможно, но смысла ревьюить yaml нет никакого.
- То же самое относится и к IDE.
- Консистентный стек языков и технологий означает, например, что вся логика пишется на актуальном для backend разработки языке. В нашем случае это Java или Python.
- Можете перечислить крупные компании, использующие Platformeco как единственный вариант BFF?
- Цитата из Jenkinsfile одного из проектов Platformeco: "PLEASE DO NOT MODIFY THIS FILE". А я хочу другую логику, другой функционал. В других проектах по-другому.
- Объясните, зачем нам 2 инструмента для управления API?

**Comment by 61803** [comment]
@alem355 Достаточно конкретики?

**Comment by mahdimahavalm** [comment]
1) Please refer to EA team for this
2) You can review not only yaml, but JS as well. 
3) Its your personal opinion. Python is out of tech stack in LMRU btw. Please refer to your EA & TA for alignment
4) Ingos, Rolf, set of companies in Finance and Telco. Your stuck to framework or language demonstrates narrow PoV on technologies
5) Not exactly - follow your requests and discuss it with product support
6) Again - you completely don't understand this topic and the point that this products covers different tasks and works together. Last call to join the training. Please discuss it with your manager. 

Best regards,
 Exchange Platform Team

**Comment by mahdimahavalm** [comment]
@panovst  Change of the git flow is technically possible. Please refer to Platformeco support with such request.

Best regards, 
 Exchange Platform Team

**Comment by AlexGladkov** [comment]
Я согласен с @61803 его аргументы вполне себе обоснованны и хотелось бы услышать какие-то аргументы кроме "you completely don't understand this topic and the point"

**Comment by panovst** [comment]
> @panovst Change of the git flow is technically possible. Please refer to Platformeco support with such request.
> 
> Best regards, Exchange Platform Team

Yes, I talked to support, the essence of the option is that the custom branch is tied to a specific environment for the CD. There is one inconvenient point that the main branch of Platformeco is tied to preprod (and it cannot be changed)

**Comment by GMayurLM** [comment]
@61803 , отмечу, что мы уже не раз сталкивались с ограничениями вызванными PL:
- В командах js разработчиков не хватает ( часто дополняются требования, нам придется активно менять и поддерживать )
- Scope работ для фронта в командах больше, чем для бэка, что не соответствует кадровому составу
- Разработка на знакомых технологиях для нас быстрее, Platformeco требует изучения
- Технология не ликвидна на рынке труда, демотивирует разработчиков
- В тех стеке best choice только для BFF и Integration Flow, игнорирован тех. стек JVM
- В тех стеке Adeo Platformeco отсутствует для SCDP
- Platformeco не встраивается в CI/CD
- Отсутсвие возможности разрабатывать unit тесты
- Затруднен Code Review - отсутствует diff
- Нельзя шарить протоколы коммуникации (DTO) между сервисами и Platformeco
- Отсутствуют зафиксированные открытые SLO
- Platformeco предоставляется в виде единой инфраструктурной единицы и не изолирует взаимную активность продуктовых групп ( доменов ), что может негативно сказываться на стабильности работы ACL слоя и, в худших сценариях, приводить к веерным перезапускам/падениям на стороне ACL
(edited)
